framework/include/executioners/FixedPointSolve.h

```diff
@@ -0,0 +1,287 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "SolveObject.h"
+
+// System includes
+#include <string>
+
+class FixedPointSolve;
+
+template <>
+InputParameters validParams<FixedPointSolve>();
+
+class FixedPointSolve : public SolveObject
```

Comment:
Note that this name is very similar as the one in test `FixedPoint`.

Reply:
I think it's fine for test objects to have close names. We are not encouraging their use